docs/template_plugin/tests/functional/op_reference/matrix_nms.cpp

```diff
@@ -0,0 +1,497 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "openvino/op/matrix_nms.hpp"
+#include "openvino/op/constant.hpp"
+#include "base_reference_test.hpp"
+
+using namespace reference_tests;
+using namespace ov;
+
+namespace {
+struct MatrixNmsParams {
+    MatrixNmsParams(
+        const int nms_top_k,
+        const float score_threshold,
+        const op::v8::MatrixNms::SortResultType sort_result_type,
+        const int keep_top_k,
+        const int background_class,
+        const op::v8::MatrixNms::DecayFunction decay_function,
+        const float gaussian_sigma,
+        const float post_threshold,
+        const ov::element::Type output_type,
+        const bool sort_result_across_batch,
+        const bool normalized,
+        const Tensor& boxes, const Tensor& scores,
+        const Tensor& expectedSelectedScores, const Tensor& expectedSelectedIndices,
+        const Tensor& expectedValidOutputs, const std::string& testcaseName = "") :
+        nms_top_k(nms_top_k),
+        score_threshold(score_threshold),
+        sort_result_type(sort_result_type),
+        keep_top_k(keep_top_k),
```

Comment:
Can we use MatrixNMS::Attributes instead of this set of parameters?

Reply:
I tried it but  MatrixNMS::Attributes has no constructor so it's almost same as this set of parameters.

Reply:
The function will have less number of arguments. So I propose to use the structure

Reply:
As MatrixNMS::Attributes needs a constructor, use local Attributes structure.

Reply:
Can we use initializer list to init operation Attribtes?